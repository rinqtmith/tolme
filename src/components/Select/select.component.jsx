import React from 'react';
import { Dropdown } from 'semantic-ui-react';

// Simple dropdown component, props: name for name, label for placeholder,
// standard for options, handleChanges for onChange, show for disable,
// value for value.
const SelectMe = ({ name, label, standard, handleChanges, show, value }) => (
  <Dropdown
    fluid
    search
    selection
    clearable
    selectOnBlur={false}
    name={name}
    placeholder={`Select a ${label}... `}
    style={{ marginBottom: '1em' }}
    options={standard}
    onChange={(event, data) => handleChanges(event, data)}
    disabled={!show}
    value={value}
  />
);

export default SelectMe;
