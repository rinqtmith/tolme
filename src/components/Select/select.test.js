import React from 'react';
import { shallow } from 'enzyme';

import Select from './select.component';
import { Dropdown } from 'semantic-ui-react';

describe('Select Component', () => {
  let wrapper;
  beforeEach(() => {
    const mockProps = {
      name: 'class',
      label: 'class label',
      standard: [
        {
          key: 1,
          value: 'some value',
          text: 'some text',
        },
        {
          key: 2,
          value: 'some value',
          text: 'some text',
        },
      ],
      handleChanges: jest.fn(),
      show: true,
      value: 'some value',
    };

    wrapper = shallow(<Select {...mockProps} />);
  });

  it('should render component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should pass event and data to given function', () => {
    expect(wrapper.find(Dropdown).simulate('change')).toBeDefined();
  });
});
