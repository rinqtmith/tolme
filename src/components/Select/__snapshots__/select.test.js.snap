// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Component should render component 1`] = `
<Dropdown
  additionLabel="Add "
  additionPosition="top"
  clearable={true}
  closeOnBlur={true}
  closeOnEscape={true}
  deburr={false}
  disabled={false}
  fluid={true}
  icon="dropdown"
  minCharacters={1}
  name="class"
  noResultsMessage="No results found."
  onChange={[Function]}
  openOnFocus={true}
  options={
    Array [
      Object {
        "key": 1,
        "text": "some text",
        "value": "some value",
      },
      Object {
        "key": 2,
        "text": "some text",
        "value": "some value",
      },
    ]
  }
  placeholder="Select a class label... "
  renderLabel={[Function]}
  search={true}
  searchInput="text"
  selectOnBlur={false}
  selectOnNavigation={true}
  selection={true}
  style={
    Object {
      "marginBottom": "1em",
    }
  }
  value="some value"
  wrapSelection={true}
/>
`;
